railties/test/application/middleware/cookies_test.rb

```diff
@@ -94,6 +94,7 @@ def read_raw_cookie
 
         config.action_dispatch.signed_cookie_digest = "SHA512"
         config.action_dispatch.signed_cookie_salt = "sha512 salt"
+        config.action_dispatch.cookies_serializer = :marshal
```

Comment:
The fact I had to make these changes to tests makes me wonder how these tests work if you are using the default `:json` serializer. I checked on `main`, and `Rails.application.config.action_dispatch.cookies_serializer` is `nil` when the requests are made. If you change it to `:json` it crashes.

Reply:
@rafaelfranca while this is fresh in mind, do you have any thoughts on this?

Reply:
@ghiculescu You'd need to pass `serializer: JSON`, to the `MessageEncryptor` and `MessageVerifier`.